### PR TITLE
Traverse PCIe topology to find optimal Max Payload Size

### PR DESCRIPTION
**Pull request by miczyg1**
Dmesg before:

```
[    2.144436] pci 0000:00:1c.0: PCI bridge to [bus 01-06]                                                                                                                                                                                     
[    2.152334] pci 0000:00:1c.0:   bridge window [mem 0x91000000-0x917fffff]                                                                                                                                                                   
[    2.156400] pci 0000:02:00.0: [1b21:1806] type 01 class 0x060400                                                                                                                                                                            
[    2.164395] pci 0000:02:00.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.172334] pci 0000:02:00.0: enabling Extended Tags                                                                                                                                                                                        
[    2.176425] pci 0000:02:00.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.180528] pci 0000:02:02.0: [1b21:1806] type 01 class 0x060400                                                                                                                                                                            
[    2.188395] pci 0000:02:02.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.196332] pci 0000:02:02.0: enabling Extended Tags                                                                                                                                                                                        
[    2.200425] pci 0000:02:02.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.204519] pci 0000:02:06.0: [1b21:1806] type 01 class 0x060400                                                                                                                                                                            
[    2.212395] pci 0000:02:06.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.220332] pci 0000:02:06.0: enabling Extended Tags                                                                                                                                                                                        
[    2.224425] pci 0000:02:06.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.232520] pci 0000:02:0e.0: [1b21:1806] type 01 class 0x060400                                                                                                                                                                            
[    2.236395] pci 0000:02:0e.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.244332] pci 0000:02:0e.0: enabling Extended Tags                                                                                                                                                                                        
[    2.248435] pci 0000:02:0e.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.256522] pci 0000:01:00.0: PCI bridge to [bus 02-06]                                                                                                                                                                                     
[    2.260338] pci 0000:01:00.0:   bridge window [mem 0x91000000-0x917fffff]                                                                                                                                                                   
[    2.268437] pci 0000:03:00.0: [8086:15f3] type 00 class 0x020000                                                                                                                                                                            
[    2.272372] pci 0000:03:00.0: reg 0x10: [mem 0x91000000-0x910fffff]                                                                                                                                                                         
[    2.280387] pci 0000:03:00.0: reg 0x1c: [mem 0x91100000-0x91103fff]                                                                                                                                                                         
[    2.284399] pci 0000:03:00.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.292473] pci 0000:03:00.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.300612] pci 0000:02:00.0: PCI bridge to [bus 03]                                                                                                                                                                                        
[    2.304338] pci 0000:02:00.0:   bridge window [mem 0x91000000-0x911fffff]                                                                                                                                                                   
[    2.312436] pci 0000:04:00.0: [8086:15f3] type 00 class 0x020000                                                                                                                                                                            
[    2.316371] pci 0000:04:00.0: reg 0x10: [mem 0x91200000-0x912fffff]                                                                                                                                                                         
[    2.324387] pci 0000:04:00.0: reg 0x1c: [mem 0x91300000-0x91303fff]                                                                                                                                                                         
[    2.328399] pci 0000:04:00.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.336472] pci 0000:04:00.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.344609] pci 0000:02:02.0: PCI bridge to [bus 04]                                                                                                                                                                                        
[    2.348338] pci 0000:02:02.0:   bridge window [mem 0x91200000-0x913fffff]                                                                                                                                                                   
[    2.356433] pci 0000:05:00.0: [8086:15f3] type 00 class 0x020000                                                                                                                                                                            
[    2.360371] pci 0000:05:00.0: reg 0x10: [mem 0x91400000-0x914fffff]                                                                                                                                                                         
[    2.368387] pci 0000:05:00.0: reg 0x1c: [mem 0x91500000-0x91503fff]                                                                                                                                                                         
[    2.372399] pci 0000:05:00.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.380473] pci 0000:05:00.0: PME# supported from D0 D3hot D3cold                                                                                                                                                                           
[    2.388609] pci 0000:02:06.0: PCI bridge to [bus 05]                                                                                                                                                                                        
[    2.392338] pci 0000:02:06.0:   bridge window [mem 0x91400000-0x915fffff]                                                                                                                                                                   
[    2.400432] pci 0000:06:00.0: [8086:15f3] type 00 class 0x020000                                                                                                                                                                            
[    2.404371] pci 0000:06:00.0: reg 0x10: [mem 0x91600000-0x916fffff]                                                                                                                                                                         
[    2.412387] pci 0000:06:00.0: reg 0x1c: [mem 0x91700000-0x91703fff]                                                                                                                                                                         
[    2.416399] pci 0000:06:00.0: Max Payload Size set to 128 (was 512, max 512)                                                                                                                                                                
[    2.424473] pci 0000:06:00.0: PME# supported from D0 D3hot D3cold                  
```

Dmesg after:

```
[    2.143488] pci 0000:00:1c.0: PCI bridge to [bus 01-06]
[    2.151388] pci 0000:00:1c.0:   bridge window [mem 0x91000000-0x917fffff]
[    2.155449] pci 0000:02:00.0: [1b21:1806] type 01 class 0x060400
[    2.163449] pci 0000:02:00.0: enabling Extended Tags
[    2.167481] pci 0000:02:00.0: PME# supported from D0 D3hot D3cold
[    2.175584] pci 0000:02:02.0: [1b21:1806] type 01 class 0x060400
[    2.179449] pci 0000:02:02.0: enabling Extended Tags
[    2.183479] pci 0000:02:02.0: PME# supported from D0 D3hot D3cold
[    2.191575] pci 0000:02:06.0: [1b21:1806] type 01 class 0x060400
[    2.195449] pci 0000:02:06.0: enabling Extended Tags
[    2.203479] pci 0000:02:06.0: PME# supported from D0 D3hot D3cold
[    2.207573] pci 0000:02:0e.0: [1b21:1806] type 01 class 0x060400
[    2.215449] pci 0000:02:0e.0: enabling Extended Tags
[    2.219482] pci 0000:02:0e.0: PME# supported from D0 D3hot D3cold
[    2.227569] pci 0000:01:00.0: PCI bridge to [bus 02-06]
[    2.231393] pci 0000:01:00.0:   bridge window [mem 0x91000000-0x917fffff]
[    2.239486] pci 0000:03:00.0: [8086:15f3] type 00 class 0x020000
[    2.243426] pci 0000:03:00.0: reg 0x10: [mem 0x91000000-0x910fffff]
[    2.251443] pci 0000:03:00.0: reg 0x1c: [mem 0x91100000-0x91103fff]
[    2.255587] pci 0000:03:00.0: PME# supported from D0 D3hot D3cold
[    2.263666] pci 0000:02:00.0: PCI bridge to [bus 03]
[    2.267393] pci 0000:02:00.0:   bridge window [mem 0x91000000-0x911fffff]
[    2.275490] pci 0000:04:00.0: [8086:15f3] type 00 class 0x020000
[    2.279426] pci 0000:04:00.0: reg 0x10: [mem 0x91200000-0x912fffff]
[    2.287442] pci 0000:04:00.0: reg 0x1c: [mem 0x91300000-0x91303fff]
[    2.295579] pci 0000:04:00.0: PME# supported from D0 D3hot D3cold
[    2.299665] pci 0000:02:02.0: PCI bridge to [bus 04]
[    2.303393] pci 0000:02:02.0:   bridge window [mem 0x91200000-0x913fffff]
[    2.311492] pci 0000:05:00.0: [8086:15f3] type 00 class 0x020000
[    2.319426] pci 0000:05:00.0: reg 0x10: [mem 0x91400000-0x914fffff]
[    2.323442] pci 0000:05:00.0: reg 0x1c: [mem 0x91500000-0x91503fff]
[    2.331578] pci 0000:05:00.0: PME# supported from D0 D3hot D3cold
[    2.335660] pci 0000:02:06.0: PCI bridge to [bus 05]
[    2.343393] pci 0000:02:06.0:   bridge window [mem 0x91400000-0x915fffff]
[    2.347487] pci 0000:06:00.0: [8086:15f3] type 00 class 0x020000
[    2.355426] pci 0000:06:00.0: reg 0x10: [mem 0x91600000-0x916fffff]
[    2.363445] pci 0000:06:00.0: reg 0x1c: [mem 0x91700000-0x91703fff]
[    2.367577] pci 0000:06:00.0: PME# supported from D0 D3hot D3cold
```

The only affected was PT601 which has the PCIe switch.

### CODE REVIEW

**Comment by miczyg1** [comment]
```
iPXE> dhcp net0
Configuring (net0 00:e0:8c:b5:d1:bd)...... ok
```

iPXE also works

**Comment by miczyg1** [comment]
The PR also fixes PCIe RP2 which should be enabled for 4-port platform, otherwise only 3 NICs are present.

File: src/mainboard/protectli/vault_jsl/mainboard.c
Comment (mkopec) [inline src/mainboard/protectli/vault_jsl/mainboard.c:33]:
```suggestion
	/* Set Max Payload to 256B */
```

**Review by mkopec — CHANGES_REQUESTED** [review]
One typo :)

**Comment by miczyg1** [comment]
> One typo :)

Done

**Comment by macpijan** [comment]
@miczyg1 Is it expected that CI fails?
@mkopec Please approve if no more comments.

**Review by mkopec — APPROVED** [review]

**Comment by miczyg1** [comment]
> @miczyg1 Is it expected that CI fails? @mkopec Please approve if no more comments.

CI fails because a PR to protectli-blobs awaits for your approval.

**Comment by macpijan** [comment]
@miczyg1 PR do blobs has been merged

**Comment by miczyg1** [comment]
> @miczyg1 PR to blobs has been merged. Do we need to adjust some code, or just re-run the job? I have just restarted it,

Rerunning will suffice. Now all checks pass.

**Pull request merged**
